src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -67,10 +69,12 @@ protected ApplicationEngine(TriggerType trigger, IVerifiable container, StoreVie
             this.ScriptContainer = container;
             this.Snapshot = snapshot;
             this.gas_amount = gas;
+            this.feeRatio = snapshot is null ? 1 : NativeContract.Policy.GetFeeRatio(Snapshot);
```

Comment:
The verifications always use the ratio `1`?

Reply:
This serves scenarios such as `Transaction.VerifyIndependent` where input snapshot is null. In such cases we should use minimum fee ratio. This is because if fee ratio is really set to 1, using other values might drop transactions which are actually valid.

Reply:
But `1` will cause attacks.

Reply:
> But `1` will cause attacks.

In PR #2054 we will check whether a witness's VerificationScript is StandardContract, so that attack should be avoided.